src/elements/Checkbox/Checkbox.tsx

```diff
@@ -1,120 +1,108 @@
-                            ${checkboxState === CheckboxSelectionState.Unselected ? "bg-transparent" : "bg-black-40"}`
-                        : checkboxState === CheckboxSelectionState.Unselected
-                        ? `hover:bg-black-5 dark:hover:bg-black-90 focus-visible:outline-violet ${unselectedStyleClasses[style]}`
-                        : `focus-visible:outline-violet ${unselectedStyleClasses[style]} ${selectedStyleClasses[style]}`
-                }`}
-                tabIndex={disabled ? -1 : 0}
-                data-test-id="checkbox"
-                aria-disabled={disabled}
-                aria-checked={
-                    checkboxState === CheckboxSelectionState.Indeterminate
-                        ? "mixed"
-                        : checkboxState === CheckboxSelectionState.Selected
-                }
-            >
-                {checkboxState === CheckboxSelectionState.Selected && <IconCheck />}
-                {checkboxState === CheckboxSelectionState.Indeterminate && <IconMinus />}
-            </span>
-            {label && (
-                <InputLabel htmlFor={id} required={required} disabled={disabled} tooltip={tooltip}>
-                    {label}
-                </InputLabel>
-            )}
+        <div className="flex flex-col gap-1 transition-colors" data-test-id="checkbox">
+            <label className="flex items-center gap-2">
+                <input {...inputProps} {...focusProps} ref={ref} className="sr-only" />
+                <span
+                    aria-hidden="true"
+                    className={merge([
+                        "relative flex w-4 h-4 items-center justify-center rounded border",
+                        isFocusVisible && "outline-violet",
+                        disabled
+                            ? merge([
+                                  "text-white pointer-events-none",
+                                  !isCheckedOrMixed(checked) &&
+                                      "border-black-20 bg-white dark:border-black-80 dark:bg-black-90",
+                                  isCheckedOrMixed(checked) &&
+                                      "border-black-40 bg-black-40 dark:border-black-60 dark:bg-black-60",
+                              ])
+                            : merge([
+                                  !isCheckedOrMixed(checked) && styles.unchecked[style],
+                                  isCheckedOrMixed(checked) && styles.checked[style],
+                              ]),
+                    ])}
+                >
+                    {checked === CheckboxState.Checked && <IconCheck />}
+                    {checked === CheckboxState.Mixed && <IconMinus />}
+                </span>
+                {label && (
+                    <span
+                        className={merge([disabled ? "text-black-40" : "text-black", "font-sans text-s font-medium"])}
+                    >
+                        {label}
+                    </span>
```

Comment:
Any reason why we don't use the `InputLabel` anymore?
Now we miss the possibility to have `tooltip` and the `htmlFor` is not bound anymore.